fastlane/lib/fastlane/actions/build_app.rb

```diff
@@ -1,13 +1,160 @@
+        absolute_dsym_path = absolute_output_path.gsub(".pkg", ".dSYM.zip") if File.extname(absolute_output_path) == ".pkg"
+
+        # This might be the mac app path, so we don't want to set it here
+        # https://github.com/fastlane/fastlane/issues/5757
+        if absolute_output_path.include?(".ipa")
+          Actions.lane_context[SharedValues::IPA_OUTPUT_PATH] = absolute_output_path
+          ENV[SharedValues::IPA_OUTPUT_PATH.to_s] = absolute_output_path # for deliver
+        elsif absolute_output_path.include?(".pkg")
+          Actions.lane_context[SharedValues::PKG_OUTPUT_PATH] = absolute_output_path
+          ENV[SharedValues::PKG_OUTPUT_PATH.to_s] = absolute_output_path # for deliver
+        end
+
+        Actions.lane_context[SharedValues::DSYM_OUTPUT_PATH] = absolute_dsym_path if absolute_dsym_path && File.exist?(absolute_dsym_path)
+        Actions.lane_context[SharedValues::XCODEBUILD_ARCHIVE] = Gym::BuildCommandGenerator.archive_path
+        ENV[SharedValues::DSYM_OUTPUT_PATH.to_s] = absolute_dsym_path if absolute_dsym_path && File.exist?(absolute_dsym_path)
+
+        return absolute_output_path
+      end
+      # rubocop:enable Metrics/PerceivedComplexity
 
       def self.description
-        "Alias for the `build_ios_app` action"
+        "Easily build and sign your app (via _gym_)"
+      end
+
+      def self.details
+        "More information: https://fastlane.tools/gym"
+      end
+
+      def self.output
+        [
+          ['IPA_OUTPUT_PATH', 'The path to the newly generated ipa file'],
+          ['PKG_OUTPUT_PATH', 'The path to the newly generated pkg file'],
+          ['DSYM_OUTPUT_PATH', 'The path to the dSYM files'],
+          ['XCODEBUILD_ARCHIVE', 'The path to the xcodebuild archive']
+        ]
+      end
+
+      def self.return_value
+        "The absolute path to the generated ipa file"
+      end
+
+      def self.author
+        "KrauseFx"
+      end
+
+      def self.available_options
+        require 'gym'
+        Gym::Options.available_options
+      end
+
+      def self.is_supported?(platform)
+        [:ios, :mac].include?(platform)
+      end
+
+      def self.example_code
+        [
+          'build_app(scheme: "MyApp", workspace: "MyApp.xcworkspace")',
+          'build_app(
+            workspace: "MyApp.xcworkspace",
+            configuration: "Debug",
+            scheme: "MyApp",
+            silent: true,
+            clean: true,
+            output_directory: "path/to/dir", # Destination directory. Defaults to current directory.
+            output_name: "my-app.ipa",       # specify the name of the .ipa file to generate (including file extension)
+            sdk: "iOS 11.1"                  # use SDK as the name or path of the base SDK when building the project.
+          )',
+          'gym              # alias for "build_app"',
+          'build_ios_app    # alias for "build_app"',
```

Comment:
`# alias for "build_app" but only iOS options`
`# alias for "build_app" but only macOS options`